param/parameterized.py

```diff
@@ -3325,27 +3332,35 @@ def __param_inheritance(mcs, param_name, param):
                 type_change = True
         del slots['instantiate']
 
-        callables = {}
+        callables, slot_values = {}, {}
+        slot_overridden = False
         for slot in slots.keys():
-            superclasses = iter(supers)
-
             # Search up the hierarchy until param.slot (which has to
             # be obtained using getattr(param,slot)) is not Undefined, or
             # we run out of classes to search.
-            while getattr(param,slot) is Undefined:
-                try:
-                    param_super_class = next(superclasses)
-                except StopIteration:
+            for scls in supers:
+                # Class may not define parameter or slot might not be
+                # there because could be a more general type of Parameter
+                new_param = scls.__dict__.get(param_name)
+                if new_param is None or not hasattr(new_param, slot):
+                    continue
+
+                new_value = getattr(new_param, slot)
+                old_value = slot_values.get(slot, Undefined)
+                if new_value is Undefined or new_value is old_value:
+                    continue
+                elif old_value is Undefined:
+                    slot_values[slot] = new_value
+                    # If we already know we have to re-validate abort
+                    # early to avoid costly lookups
+                    if slot_overridden or type_change:
+                        continue
```

Comment:
As far as I can see this `continue` is not skipping any code path, was it supposed to be a `break`?